lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { buildClientInfo, ClientInfo } from "../account/ClientInfo";
+import { ICrypto } from "../crypto/ICrypto";
+import { ICacheStorage } from "../cache/ICacheStorage";
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringUtils } from "../utils/StringUtils";
+import { ServerAuthorizationCodeResponse } from "../server/ServerAuthorizationCodeResponse";
+import { Logger } from "../logger/Logger";
+import { ServerError } from "../error/ServerError";
+import { IdToken } from "../account/IdToken";
+import { UnifiedCacheManager } from "../unifiedCache/UnifiedCacheManager";
+import { ScopeSet } from "../request/ScopeSet";
+import { TimeUtils } from "../utils/TimeUtils";
+import { TokenCacheGenerator } from "../unifiedCache/generator/TokenCacheGenerator";
+import { AuthenticationResult } from "./AuthenticationResult";
+import { AccountEntity } from "../unifiedCache/entities/AccountEntity";
+import { Authority } from "../authority/Authority";
+import { AuthorityType } from "../authority/AuthorityType";
+
+/**
+ * Class that handles response parsing.
+ */
+export class ResponseHandler {
+    private clientId: string;
+    private uCacheManager: UnifiedCacheManager;
+    private cryptoObj: ICrypto;
```

Comment:
nit: I think we could be more precise with naming. Maybe cryptoUtils?

Reply:
Will remember this.